### PR TITLE
option for online player required when backing up

### PR DESCRIPTION
**Pull request by kafuuchino-desu**
#4 redo, fixed some logic and hopefully I didn't screw up this time.

### CODE REVIEW

**Comment by Fallen-Breath** [comment]
I'll suggest to apply the online player requirement logic only when tqm is loaded at server startup

You can do that via e.g. adding a global flag and set the flag to true only in `on_server_start`. Then your all new stuffs are enabled iif the flag is true

If tqm is loaded when the server is already running, the current logic may maintain a wrong `playercount` which will lead to unintended behavior

**Comment by kafuuchino-desu** [comment]
> I'll suggest to apply the online player requirement logic only when tqm is loaded at server startup
> 
> You can do that via e.g. adding a global flag and set the flag to true only in `on_server_start`. Then your all new stuffs are enabled iif the flag is true
> 
> If tqm is loaded when the server is already running, the current logic may maintain a wrong `playercount` which will lead to unintended behavior

thank you for the advice <3, the logic should be fine now. (btw I'm new to MCDR plugin dev so sorry for the bad code XD)

File: timed_quick_backup_multi/entry.py [hunk 1]
```diff
@@ -90,6 +92,32 @@ def on_load(server: PluginServerInterface, prev):
...
+		clock.set_enabled(False)
+	else:
+		clock.set_enabled(config.enabled)
+
+
+def on_player_joined(server: PluginServerInterface, player: str, info: Info):
+	global playercount, config, clock
+	if playercount == 0:
+		playercount += 1
+		clock.set_enabled(config.enabled)
```
Comment (Fallen-Breath) [inline timed_quick_backup_multi/entry.py]: No config value check here

File: timed_quick_backup_multi/entry.py [hunk 2]
```diff
@@ -90,6 +92,32 @@ def on_load(server: PluginServerInterface, prev):
...
+	else:
+		playercount += 1
+
+
+def on_player_left(server: PluginServerInterface, player: str, info: Info):
+	global playercount, config, clock
+	playercount -= 1
+	if playercount == 0:
+		clock.set_enabled(False)
```
Comment (Fallen-Breath) [inline timed_quick_backup_multi/entry.py]: No config value check here too, also the clock will be disabled even if plugin is loaded midway in the server's lifecyle

**Comment by kafuuchino-desu** [comment]
I think this is the best I can think of, loading tqb fresh after server is started will be bugged, but I didn't found MCDR provides a API for me to get online players count. This could be potentially fixed parsing /list command tho.

**Comment by Fallen-Breath** [comment]
> I think this is the best I can think of, loading tqb fresh after server is started will be bugged, but I didn't found MCDR provides a API for me to get online players count. This could be potentially fixed parsing /list command tho.

I have given an example workround using a global flag indicating if the plugin has survived since a server startup in the commet above https://github.com/TISUnion/TimedQBM/pull/5#issuecomment-1325383197

Yeah MCDR doesn't provide such player count API, [this api](https://github.com/MCDReforged/MinecraftDataAPI) does btw

**Comment by kafuuchino-desu** [comment]
> > I think this is the best I can think of, loading tqb fresh after server is started will be bugged, but I didn't found MCDR provides a API for me to get online players count. This could be potentially fixed parsing /list command tho.
> 
> I have given an example workround using a global flag indicating if the plugin has survived since a server startup in the commet above [#5 (comment)](https://github.com/TISUnion/TimedQBM/pull/5#issuecomment-1325383197)
> 
> Yeah MCDR doesn't provide such player count API, [this api](https://github.com/MCDReforged/MinecraftDataAPI) does btw

Oh I see what you mean now, sorry for the bother, hopefully the logic is ok now

**Comment by Fallen-Breath** [comment]
The changes is still a bit messy, e.g. clock state sync everywhere and incompleted if-statements. I'll make a commit to fix them so it's easier for both of us

**Pull request merged**